framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2504,8 +2504,18 @@ protected boolean checkMethodReferenceAsOverride(
         // That is handled separately in method receiver check.
 
         // The type of the expression or type use, <expression>::method or <type use>::method.
-        AnnotatedTypeMirror enclosingType =
-                atypeFactory.getAnnotatedType(memberReferenceTree.getQualifierExpression());
+        ExpressionTree qexpTree = memberReferenceTree.getQualifierExpression();
+        AnnotatedTypeMirror enclosingType = atypeFactory.getAnnotatedTypeFromTypeTree(qexpTree);
+        if (qexpTree.getKind() == Tree.Kind.IDENTIFIER) {
+            // This is a hack around the ambiguity between identifiers that could be
+            // types or expressions. Using TreeUtils.isTypeTree unfortunately doesn't
+            // help, as it isn't precise enough.
+            // TODO: determine a nicer way. Is taking the main qualifiers enough?
```

Comment:
I think the correct way is to check the `kind` field of the member reference.  I tried this here:
https://github.com/typetools/checker-framework/compare/master...smillst:smillst1809?expand=1  And it seems to work with out changing the ATF caches.

Reply:
Ah, I only looked in MemberReferenceTree not in JCMemberReference. Thanks, I'll change this.